internal/mysqld/executor/executor.go

```diff
@@ -129,6 +140,242 @@ func (e *defaultExecutor) execSelect(ctx context.Context, n *planner.NodeSelectS
 	return wrapQueryResults(res), nil
 }
 
+func (e *defaultExecutor) execANNS(ctx context.Context, q *planner.NodeQuerySpecification, outputs []string) (*sqltypes.Result, error) {
+	if !q.Limit.IsSome() {
+		return nil, fmt.Errorf("limit not specified in the ANNS statement")
+	}
+	limit := q.Limit.Unwrap().Limit
+	offset := q.Limit.Unwrap().Offset
+
+	annsClause := q.Anns.Unwrap()
+
+	searchParams := map[string]string{
+		"anns_field": annsClause.Column.Name,
+		"topk":       fmt.Sprintf("%d", limit),
+		"offset":     fmt.Sprintf("%d", offset),
+	}
+
+	if annsClause.Params.IsSome() {
+		for k, v := range annsClause.Params.Unwrap().KVs {
+			searchParams[k] = v
+		}
+		params, _ := json.Marshal(annsClause.Params.Unwrap().KVs)
+		searchParams["params"] = string(params)
+	}
+
+	outputsIndex, userOutputs := generateOutputsIndex(outputs)
+
+	_, queryNumberExist := outputsIndex["query_number"]
+	if !queryNumberExist {
+		return nil, fmt.Errorf("query_number must be specified in the ANNS statement")
+	}
+
+	filter := ""
+	from := q.From.Unwrap()
+	if from.Where.IsSome() {
+		filter = planner.NewExprTextRestorer().RestoreExprText(from.Where.Unwrap())
+	}
+
+	tableName := from.TableSources[0].TableName.Unwrap()
+
+	phg := vector2PlaceholderGroupBytes(annsClause.Vectors)
+
+	req := &milvuspb.SearchRequest{
+		CollectionName:     tableName,
+		PartitionNames:     nil,
+		Dsl:                filter,
+		PlaceholderGroup:   phg,
+		DslType:            commonpb.DslType_BoolExprV1,
+		OutputFields:       userOutputs,
+		SearchParams:       mapKvPairs(searchParams),
+		TravelTimestamp:    0,
+		GuaranteeTimestamp: 2, // default bounded consistency level.
+		Nq:                 int64(len(annsClause.Vectors)),
+	}
+
+	res, err := e.s.Search(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	if res.GetStatus().GetErrorCode() != commonpb.ErrorCode_Success {
+		return nil, common.NewStatusError(res.GetStatus().GetErrorCode(), res.GetStatus().GetReason())
+	}
+
+	return wrapSearchResult(res, outputsIndex, userOutputs)
+}
+
+func wrapSearchResult(res *milvuspb.SearchResults, outputsIndex map[string]int, userOutputs []string) (*sqltypes.Result, error) {
+	qnIndex := outputsIndex["query_number"]
```

Comment:
by design , query_number is also optional, just like distance.
This part can be discussed again.